docs/source/core_developers_guide/environment_setup.rst

```diff
@@ -0,0 +1,374 @@
+  or later)
+
+These tools are available for the following platforms:
+
+* Linux
+* Windows
+* macOS
+
+
+Vagrant
+-------
+
+Vagrant is a tool which installs and configures virtual development
+environments. It allows development teams to easily specify and share
+consistent virtual machine configurations.
+
+The dev tools directory (path ''sawtooth-core/tools'') of the sawtooth-core
+repository contains a Vagrant configuration which is specifically tailored to
+Sawtooth development.  A new developer with installed copies of Vagrant and
+VirtualBox can clone the sawtooth-core repository and have a functional VM
+which can run validators within a few minutes.
+
+A quick introduction to using dev tools is available as a
+quickstart in this Developer's Guide.
+
+
+Layout of Dev Tools 
+===================
+
+The dev tools directory is structured as follows:
+
+.. code-block:: console
+
+ tools/
+    bootstrap.d/
+    guest-files/
+    plugins/
+    scripts/
+    tests/
+    win-bin/
+    Vagrantfile
+
+
+Vagrantfile
+  Vagrantfile is the main configuration file for Vagrant.  It is ruby
+  code executed by Vagrant and is executed every time a Vagrant command is
+  executed.
+
+bootstrap.d
+  The bootstrap.d directory contains a set of bash scripts which are
+  executed in order during the provisioning step of 'vagrant up'.  These
+  scripts are always executed.
+
+guest-files
+  The guest-files directory contains configuration files which are used by
+  the bootstrap.d scripts.
+
+  There is also a local-env.sh script which contains environment specific
+  variables for Sawtooth  development, such as PYTHONPATH.
+
+plugins
+  The plugins directory contains bash scripts which can be easily configured
+  to execute during the provisioning step of 'vagrant up'.  These scripts
+  run after bootstrap.d scripts.
+
+scripts
+  This scripts directory contains scripts which are sometimes useful to the
+  developer after provisioning has been completed and the developer has a
+  shell in the virtual machine.  For example, there are scripts which
+  help build Ubuntu packages.
+
+tests
+  The tests directory includes tests run within the Vagrant environment. These are
+  in addition to the unit and integration tests found in the tests directory of
+  sawtooth-core and sawtooth-validator.
+
+win-bin
+  The win-bin directory includes scripts for running Sawtooth natively
+  under Windows and is not used in the Vagrant environment.
+
+
+Layout in the Virtual Machine
+-----------------------------
+
+A convenient feature of Vagrant is the easy sharing of directories between
+the host machine and the guest virtual machine. By default the Vagrant
+configuration directory is mounted under /vagrant in the virtual machine.
+
+In the dev tools configuration a /project mount point is also
+defined which provides access to the Sawtooth repositories.
+
+
+Configuration Options
+=====================
```

Comment:
Agreed. Let's add a story if we think we need this re-added with updated info. I'll make a to-do for this for early next week.